### PR TITLE
Implement molecular-profile-sample-counts endpoint

### PR DESCRIPTION
**Pull request by alisman**

### CODE REVIEW

File: src/main/java/org/cbioportal/persistence/StudyViewRepository.java [hunk 1]
```diff
@@ -1,9 +1,6 @@
 package org.cbioportal.persistence;
 
-import org.cbioportal.model.AlterationCountByGene;
-import org.cbioportal.model.ClinicalData;
-import org.cbioportal.model.ClinicalDataCount;
-import org.cbioportal.model.Sample;
+import org.cbioportal.model.*;
```
Comment (alisman) [inline src/main/java/org/cbioportal/persistence/StudyViewRepository.java:3]: i think IDE did this.  what's right?

Comment (haynescd) [reply src/main/java/org/cbioportal/persistence/StudyViewRepository.java:3]: Do you need help with your ide settings?

File: src/main/java/org/cbioportal/persistence/mybatisclickhouse/StudyViewMyBatisRepository.java [hunk 2]
```diff
@@ -41,6 +40,15 @@ public List<ClinicalDataCount> getClinicalDataCounts(StudyViewFilter studyViewFi
             filteredAttributes, FILTERED_CLINICAL_ATTR_VALUES );
     }
 
+    @Override
+    public List<GenomicDataCount> getGenomicDataCounts(StudyViewFilter studyViewFilter, CategorizedClinicalDataCountFilter categorizedClinicalDataCountFilter) {
+
+        //return new ArrayList<GenomicDataCount>();
```
Comment (haynescd) [inline src/main/java/org/cbioportal/persistence/mybatisclickhouse/StudyViewMyBatisRepository.java:46]: Remove

File: src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewMapper.xml [hunk 3]
```diff
@@ -104,6 +104,19 @@
             <property name="table_name_prefix" value="sample"/>
         </include>
     </select>
+
+
+    <!-- for /molecular-profile-sample-counts/fetch (returns GenomicDataCount) -->
+    <select id="getGenomicDataCounts" resultType="org.cbioportal.model.GenomicDataCount">
+        SELECT genetic_profile.stable_id AS value, genetic_profile.stable_id AS value2, genetic_profile.name AS label, count(sample_profile.genetic_profile_id) AS count FROM sample_profile
```
Comment (haynescd) [inline src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewMapper.xml:111]: Can we have each column on a new line? (just easier to read)
Ex. 
Select 
  stable_id,
  name

**Review by haynescd — CHANGES_REQUESTED** [review]
Looks good (Now you are a backend dev.) Just small changes

File: src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewFilterMapper.xml [hunk 4]
```diff
@@ -13,6 +13,26 @@
                     #{studyId}
                 </foreach>
             </if>
+
+
+            <if test="studyViewFilter.genomicProfiles != null and !studyViewFilter.genomicProfiles.isEmpty()">
```
Comment (alisman) [inline src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewFilterMapper.xml:18]: @gblaih here is the filter.  i found a small problem with foreach at the bottom but this is the basic idea.

**Pull request closed without merging**